### PR TITLE
Topic4

### PR DESCRIPTION
**Pull request by OctaBruzzesi**

### CODE REVIEW

File: 04-HTML5-APIs/js/canvas.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    for(i=0;i<5;i++){
+      for(r=0;r<5;r++){
+        ctx.beginPath();
+        x = 50+r*50;
+        y = 50+i*50;
+        radius = 25;
+        angleInit = 0;
+        angleEnd = Math.PI+(Math.PI*r)/2;
+        antiClockWise = i/2;
+        if(antiClockWise === 0){
```
Comment (afiorenza) [inline 04-HTML5-APIs/js/canvas.js]: Why are you doing this?

File: 04-HTML5-APIs/Exercise1.html [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        <button class="button" onclick="storeLocalTextArea()">Save localStorage</button>
+        <button class = "button" onclick="saveIDBTextArea()">Save on IDB</button>
+        <button class = "button" onclick="deleteLocalTextArea()">Clear localStorage</button>
+        <button class = "button" onclick="deleteIDBTextArea()">Clear IDB</button>
+      </div>
+      <br>
+      <textarea id="drop"> Drop here </textarea>
+      <p id="status"> <br>
+      <h3 id="socket">Socket Test</h3>
+        <p id = "a">
```
Comment (afiorenza) [inline 04-HTML5-APIs/Exercise1.html]: the space between letters is not needed here

File: 04-HTML5-APIs/js/canvasAnimated.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+function initialize () {
+  let x =  0;
+  let y = 5;
+  let drawSpeedx = 3;
+  let drawSpeedy = 4;
+
+  function animateDraw() {
+    let animationFrame = window.mRequestAnimationFrame||window.webkitRequestAnimationFrame||window.msRequestAnimationFrame||
```
Comment (afiorenza) [inline 04-HTML5-APIs/js/canvasAnimated.js]: Leave space between ||

File: 04-HTML5-APIs/js/jsExercise1.js [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+};
+holder.ondragend = function() {
+    this.className = '';
+    return false;
+};
+holder.ondrop = function(e) {
+    this.className = '';
+    e.preventDefault();
+
+    var file = e.dataTransfer.files,
```
Comment (afiorenza) [inline 04-HTML5-APIs/js/jsExercise1.js]: remember to use es6 :D

File: 04-HTML5-APIs/js/jsExercise1.js [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+  var request = result.transaction("textDataBase", "readwrite").objectStore("textDataBase").clear();
+
+  request.onsuccess = function (event) {
+    alert("Remove");
+  }
+}
+
+//Exercise 3 drop zone
+
+function allowDrop(ev){
```
Comment (afiorenza) [inline 04-HTML5-APIs/js/jsExercise1.js]: Refactor to match a more descriptive name of `ev` like `event`

File: 04-HTML5-APIs/js/jsExercise1.js [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+
+  function onMessage(evt)
+  {
+    writeToScreen('<span style="color: blue;">RESPONSE: ' + evt.data+'</span>');
+    websocket.close();
+  }
+
+  function onError(evt)
+  {
+    writeToScreen('<span style="color: red;">ERROR:</span> ' + evt.data);
```
Comment (afiorenza) [inline 04-HTML5-APIs/js/jsExercise1.js]: The style should go on a stylesheet. And add a class here.

File: 04-HTML5-APIs/js/jsExercise1.js [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+}
+
+// exercise 4 socket
+
+
+  var wsUri = "ws://echo.websocket.org/";
+  var output;
+
+  function init()
+  {
```
Comment (afiorenza) [inline 04-HTML5-APIs/js/jsExercise1.js]: The `{` should go on the above line. The same with other cases.

File: 04-HTML5-APIs/js/jsExercise1.js [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+    output = document.getElementById("output");
+    testWebSocket();
+  }
+
+  function testWebSocket()
+  {
+    websocket = new WebSocket(wsUri);
+    websocket.onopen = function(evt) { onOpen(evt) };
```
Comment (afiorenza) [inline 04-HTML5-APIs/js/jsExercise1.js]: the function here only call another function... Why don't you call the function `onOpen` directly?

File: 04-HTML5-APIs/js/jsExercise1.js [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+  }
+
+  function onClose(evt)
+  {
+    writeToScreen("DISCONNECTED");
+  }
+
+  function onMessage(evt)
+  {
+    writeToScreen('<span style="color: blue;">RESPONSE: ' + evt.data+'</span>');
```
Comment (afiorenza) [inline 04-HTML5-APIs/js/jsExercise1.js]: The style should go on a stylesheet. And add a class here.

Comment (afiorenza) [reply 04-HTML5-APIs/js/jsExercise1.js]: Also you could use an es6 template literal here

File: 04-HTML5-APIs/js/jsExercise1.js [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+
+  function doSend(message)
+  {
+    writeToScreen("SENT: " + message);
+    websocket.send(message);
+  }
+
+  function writeToScreen(message)
+  {
+    var pre = document.createElement("p");
```
Comment (afiorenza) [inline 04-HTML5-APIs/js/jsExercise1.js]: `pre` is not a descriptive name. Please refactor.

File: 04-HTML5-APIs/js/jsExercise1.js [hunk 11]
```diff
@@ -0,0 +1,153 @@
...
+      document.getElementById('drop').value = event.target.result;
+        console.log(event.target);
+    };
+    reader.readAsText(file[0],"UTF-8");
+}
+
+// exercise 4 socket
+
+
+  var wsUri = "ws://echo.websocket.org/";
```
Comment (afiorenza) [inline 04-HTML5-APIs/js/jsExercise1.js]: Remember to use es6, also is more legible if you declare global variables on the top of the file.

**Review by afiorenza — CHANGES_REQUESTED** [review]
Well done! just a few changes

File: 04-HTML5-APIs/js/jsExercise1.js (hunk 4 above)
Comment (OctaBruzzesi) [reply 04-HTML5-APIs/js/jsExercise1.js]: this.constructor.name = '';
that would be es6?

Comment (OctaBruzzesi) [reply 04-HTML5-APIs/js/jsExercise1.js]: or just put let file = ...?

**Review by afiorenza — APPROVED** [review]

**Comment by afiorenza** [comment]
Remember to fix conflict and merge it

**Pull request merged**